lib/auth/register.go

```diff
@@ -308,10 +310,14 @@ func proxyServerIsAuth(server utils.NetAddr) bool {
 // registerThroughProxy is used to register through the proxy server.
 func registerThroughProxy(token string, params RegisterParams) (*proto.Certs, error) {
 	var certs *proto.Certs
+
+	// Determine if either the local library is set to insecure, or insecure was passed as a parameter.
+	var insecure = params.Insecure || lib.IsInsecureDevMode()
```

Comment:
I'm tempted to say we should drop `lib.IsInsecureDevMode()` here - we're trying to move away from these globals and the `Register` method is only called in a handful of places. We can simply inject `lib.IsInsecureDevMode()` as `RegisterParams.Insecure` where those call sites are.

Reply:
I've removed the use of lib.IsInsecureDevMode.